### PR TITLE
Print counterexamples on variant violations and deadlocks

### PR DESCRIPTION
**Pull request by shonfeder**
Towards #888

This adds handling for variant violations and deadlocked errors reported by Apalache.

There are also tests added for CLI configuration via `--step` and `--init` flags.

- [x] Tests added for any new code
- [x] Documentation added for any new functionality
- [x] Entries added to the respective `CHANGELOG.md` for any new functionality
- [x] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

### CODE REVIEW

File: quint/src/cliCommands.ts [hunk 1]
```diff
@@ -558,10 +563,22 @@ export async function verifySpec(prev: TypecheckedStage): Promise<CLIProcedure<V
...
       .map(_ => ({ ...verifying, status: 'ok', errors: [] } as VerifiedStage))
-      .mapLeft(err => ({
-        msg: err.explanation,
-        stage: { ...verifying, status: 'failure', errors: err.errors },
-      }))
+      .mapLeft(err => {
+        const trace: QuintEx[] | undefined = err.traces ? ofItf(err.traces[0]) : undefined
+        if (trace !== undefined) {
+          // Always print the conterexample, unless the output is being directed to one of the outfiles
+          const verbosityLevel = prev.args.out || prev.args.outItf ? 0 : 2
```
Comment (thpani) [inline quint/src/cliCommands.ts:570]: Meaning of the magic constants 0 and 2 is not clear to me.
Should we have names for these?

File: quint/src/cliCommands.ts [hunk 2]
```diff
@@ -558,10 +563,22 @@ export async function verifySpec(prev: TypecheckedStage): Promise<CLIProcedure<V
...
     res
       .map(_ => ({ ...verifying, status: 'ok', errors: [] } as VerifiedStage))
-      .mapLeft(err => ({
-        msg: err.explanation,
-        stage: { ...verifying, status: 'failure', errors: err.errors },
-      }))
+      .mapLeft(err => {
+        const trace: QuintEx[] | undefined = err.traces ? ofItf(err.traces[0]) : undefined
+        if (trace !== undefined) {
+          // Always print the conterexample, unless the output is being directed to one of the outfiles
```
Comment (thpani) [inline quint/src/cliCommands.ts:569]: To double-check my understanding:
This is because on verify, we always expect a counterexample to be printed, right?

**Review by thpani — APPROVED** [review]
Looks great!

File: quint/src/cliCommands.ts (hunk 1 above)
Comment (shonfeder) [reply quint/src/cliCommands.ts:570]: Probably! This mysterious numberings is pervasive throughout this file unfortunately. I’ll make a followup to note the need to apply some cleanup to the verbosity system.

File: quint/src/cliCommands.ts (hunk 2 above)
Comment (shonfeder) [reply quint/src/cliCommands.ts:569]: That’s my assumption, yep! This is following the “spec” Igor wrote in the docs for the command: https://github.com/informalsystems/quint/blob/main/doc/quint.md#command-verify

File: quint/src/cliCommands.ts (hunk 1 above)
Comment (shonfeder) [reply quint/src/cliCommands.ts:570]: Created an issue to follow up! https://github.com/informalsystems/quint/issues/915

**Comment by shonfeder** [comment]
Thanks for the review!

File: quint/src/cliCommands.ts (hunk 1 above)
Comment (konnov) [reply quint/src/cliCommands.ts:570]: We should use `defaultLevel` of `verbosity.ts`

**Pull request merged**